pkg/metrics/metrics.go

```diff
@@ -186,10 +188,19 @@ func New(log logr.Logger, c clock.Clock) *Metrics {
 		)
 	)
 
+	// Create Registry and register the recommended collectors
```

Comment:
Suggest rewording this comment to explain **why** rather than what the following code does.
Perhaps:

> We want to avoid using and mutating global state so we avoid
> `prometheus.DefaultRegisterer`. Instead we create a local Registry and manually register
> the same two collectors that are used by DefaultRegisterer.
>
> # Process collector
>
> Exports the current state of process metrics including CPU, memory and file
> descriptor usage as well as the process start time. The detailed behavior is
> defined by the provided ProcessCollectorOpts. The zero value of
> ProcessCollectorOpts creates a collector for the current process with an empty
> namespace string and no error reporting.
>
> The collector only works on operating systems with a Linux-style proc
> filesystem and on Microsoft Windows. On other operating systems, it will not
> collect any metrics.
>
> # Go collector
>
> Exports metrics about the current Go process using debug.GCStats (base
> metrics) and runtime/metrics.